File: src/main/java/arcana/common/aspects/graph/nodes/AspectListCached.java
package arcana.common.aspects.graph.nodes;

import arcana.common.aspects.AspectList;

public class AspectListCached {
    public AspectList list = new AspectList();
    public int value = Integer.MAX_VALUE;

    public AspectListCached(){}

    public AspectListCached(AspectList aspects){
        update(aspects);
    }

    public void update(AspectList aspects){
        list = aspects;
        value = list.sum();
    }

    @Override
    public String toString() {
        return list.toString() + '|' + value;
    }
}

File: src/main/kotlin/arcana/common/aspects/graph/nodes/AspectListCached.kt
package arcana.common.aspects.graph.nodes

import arcana.common.aspects.AspectList

class AspectListCached {
    var list = AspectList()
    var value = Int.MAX_VALUE

    constructor() {}
    constructor(aspects: AspectList) {
        update(aspects)
    }

    fun update(aspects: AspectList) {
        list = aspects
        value = list.sum()
    }

    override fun toString(): String {
        return "$list|$value"
    }
}